### ISSUE TITLE
Is it possible to obtain detailed metrics using another Python library with the decision tree of chefboost?

### ISSUE THREAD

**Issue by anapaulamendes** [serengil/chefboost#7]
I saw in another issue that the detailed metrics are not yet implemented in chefboost, so I tried to use scikit learn metrics, but it was not compatible. Do you know any external alternatives at the moment?

**Comment by serengil** [comment]
Sure, I will cover this in the next release. This issue will be open until new verison is released.

If you add what metrics do you need exactly under this message, it would be helpful.

**Comment by anapaulamendes** [comment]
I know there are many, but these would be:

Model Validation:

- Cross-validation (k-fold and leave-one-out)
- Use trainig set

Metrics:

- Kappa statistic, Mean absolute error, Root mean squared error, Relative absolute error, Root relative squared error.
- Detailed Accuracy By Class:
 - TP Rate, FP Rate, Precision, Recall, F-Measure, MCC, ROC Area, PRC Area, Class
- Confusion Matrix

**Comment by bobcolner** [comment]
much better would be to just conform to the sklean API. Then you use all the metrics/model selections tools that have standardized on sklearn API. 

This liberay is great btw, fills a longstanding gap in sklean.

**Comment by anapaulamendes** [comment]
@bobcolner scikit-learn doesn't work with categorical data and this leads to unfortunate implications for problems that are restricted to working exclusively with categorical data.
Chefboost is a lib that works very well with categorical data and has a lot of potential, so it is natural that it'll have new implementations and metrics.

**Comment by serengil** [comment]
Good news! Most of those accuracy metrics are wrapped in the 0.0.7 version of chefboost. I also added an evaluate function in the chefboost interface. tests/global-unit-test.py will guide you how to run those features. thank you for your support.

**Comment by serengil** [comment]
BTW, do not hesitate to open new issues if you see any improvement :) I will be appreciate.

**Comment by anapaulamendes** [comment]
@serengil thank you so much! 😄
